### ISSUE TITLE
assertion error for infinite offset

### ISSUE THREAD

**Issue by lkapelevich** [ERGO-Code/HiGHS#1410]
I am getting the following assertion failure on latest (and HiGHS aborts):

```
void presolve::HPresolve::fixColToLower(presolve::HighsPostsolveStack&, HighsInt): Assertion `std::isfinite(model->offset_)' failed.
Aborted
```

when I pass an objective vector with infinities (which I thought is allowed).

Small example:

```cpp
int main() {
    Highs highs;
    HighsModel model;
    model.lp_.num_col_ = 3;
    model.lp_.num_row_ = 2;
    model.lp_.sense_ = ObjSense::kMinimize;
    model.lp_.offset_ = 0;
    model.lp_.col_cost_ = {-3, -2, -kHighsInf};
    // model.lp_.col_cost_ = {-3, -2, -INFINITY};
    model.lp_.col_lower_ = {0.0, 0.0, 0.0};
    model.lp_.col_upper_ = {1.0e30, 1.0e30, 1};
    model.lp_.row_lower_ = {-1.0e30, 12};
    model.lp_.row_upper_ = {7.0, 12.0};
    model.lp_.a_matrix_.format_ = MatrixFormat::kColwise;
    model.lp_.a_matrix_.start_ = {0, 2, 4, 6};
    model.lp_.a_matrix_.index_ = {0, 1, 0, 1, 0, 1};
    model.lp_.a_matrix_.value_ = {1.0, 4.0, 1.0, 2.0, 1, 1.0};

    model.lp_.integrality_.resize(model.lp_.num_col_);
    model.lp_.integrality_[0] = HighsVarType::kContinuous;
    model.lp_.integrality_[0] = HighsVarType::kContinuous;
    model.lp_.integrality_[0] = HighsVarType::kInteger;

    HighsStatus modstatus = highs.passModel(model);
    std::cout << "model status: " << (modstatus == HighsStatus::kOk) << std::endl; //  gives model status: 1

    std::cout << "running..." << std::endl;
    HighsStatus return_status = highs.run();
}

```

My interpretation of the docstring for `infinite_cost` was that it's legal to provide large values in the cost vector, they are just treated as "infinity", so I expected this to run.

**Comment by odow** [comment]
Do many solvers support infinity in cost coefficients? I had someone ask me about this on Monday, so perhaps it's a more common situation than I assumed.

I guess a simple presolve trick is to fix the bounds to `0` and update the coefficient to `0`.

**Comment by jajhall** [comment]
Our interpretation of infinite_cost is the limit above which costs are deemed to be too large and the model considered to be invalid.

However, Oscar is right that an infinite cost can be interpreted by changing bounds in presolve

**Comment by lkapelevich** [comment]
> Our interpretation of infinite_cost is the limit above which costs are deemed to be too large and the model considered to be invalid.

I see. The message in the output when those costs are passed is a warning and not an error, and `passModel` returned `kOk`. I think for other invalid data you display an error.

> That said I'm surprised to see that HiGHS has got as far as presolve in this case. I'll try to reproduce the behaviour

Thanks!

**Comment by jajhall** [comment]
Yes, costs greater than `infinite_cost` are permitted. I wanted to make them illegal, but - about 4 years ago - something made me permit them, but not modify the bounds to prevent the infinite cost from contributing to the objective. 

I'll continue to allow them but, since the upper bound on the setting of `infinite_cost` is `kHighsInf`, I'll change the interpretation so that costs are considered infinite if they are greater than _or equal to_ `infinite_cost`. Ditto for `infinite_bound` and `large_matrix_value`, for consistency.

So long as the bounds on the variable include zero, any variable with infinite cost will be fixed at zero. If zero is not feasible then there will be an error return.

**Comment by jajhall** [comment]
Since infinite costs can be interpreted, and it's interesting to do so, I'll continue to allow them. Precisely what the solver(s) do has still to be coded (and I don't have time) so, if they are given a model with infinite costs, `Highs::run()` will return `HighsStatus::kError` for now. 

After a bit of thought this morning, the "zero argument" is a bit restrictive. If a variable `x` has bounds `[l, u]` and a cost of +infinity then, in the first instance (and assuming minimization), it can be fixed to the least (integer, if it's integral) value in `[l, u]`. I'd argue that an optimal solution for these fixed values solves the original problem. If the problem is infeasible for these fixed values, then it's less clear what can be done!

**Comment by jajhall** [comment]
Closed by #1416
